src/caffe/layers/cudnn_softmax_layer.cpp

```diff
@@ -26,10 +26,10 @@ template <typename Dtype>
 void CuDNNSoftmaxLayer<Dtype>::Reshape(const vector<Blob<Dtype>*>& bottom,
       const vector<Blob<Dtype>*>& top) {
   SoftmaxLayer<Dtype>::Reshape(bottom, top);
-  int N = bottom[0]->num();
```

Comment:
I don't think this should be changed -- in our usage the cuDNN softmax is run in "channel mode" so that the softmax is done independently for each (N, H, W). This will break models expecting this behavior for softmax over spatial dimensions.

We should perhaps expose the choice of "instance mode" too.